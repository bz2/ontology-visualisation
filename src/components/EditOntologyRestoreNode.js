import { useState } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { Button } from 'primereact/button'
import { MultiSelect } from 'primereact/multiselect'
import actions from '../store/actions'
import setOntology from '../utils/editOntology/setOntology'

const EditOntologyRestoreNode = ({
  type,
  operation,
  setStoreState,
  addNumber,
  optionNodes,
  optionEdges
}) => {
  const { t } = useTranslation()

  const [selectedElement, setSelectedElement] = useState(undefined)
  const [selectedElementProperties, setSelectedElementProperties] = useState(undefined)

  return (
    <>
      <div
        className="edit-ontology-row"
      >
        {
          (type === 'node'
            && optionNodes.length === 0)
          || (type === 'edge'
            && optionEdges.length === 0)
            ? (
              <div>
                {t('noDeletedElements')}
              </div>
            ) : (
              <>
                <label htmlFor="element-select">
                  {t('selectElement')}
                </label>

                <MultiSelect
                  value={selectedElement}
                  options={type === 'node' ? optionNodes : optionEdges}
                  onChange={(e) => setSelectedElement(e.value)}
                  placeholder={t('selectElement')}
                  display="chip"
                  filter
                  showClear
                  filterBy="label"
                />
              </>
            )
        }

      </div>

      <div className="edit-ontology-row">
        <Button
          className="go-button"
          tooltip={`${t(operation)}`}
          disabled={!selectedElement}
          onClick={() => {
            setOntology({
              operation,
              type,
              selectedElement,
              setStoreState,
              addNumber,
              selectedElementProperties,
              t
            })
            setSelectedElement(undefined)
            setSelectedElementProperties({})
          }}
          label={t(operation)}
          icon="pi pi-chevron-right"
          iconPos="right"
        />
      </div>
    </>
  )
}

EditOntologyRestoreNode.propTypes = {
  type: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
  setStoreState: PropTypes.func.isRequired,
  addNumber: PropTypes.func.isRequired,
  optionNodes: PropTypes.arrayOf(PropTypes.shape).isRequired,
  optionEdges: PropTypes.arrayOf(PropTypes.shape).isRequired,
}

const mapToProps = ({
  selectedGraphVersion,
  graphVersions,
}) => ({
  selectedGraphVersion,
  graphVersions,
})

export default connect(
  mapToProps,
  actions
)(EditOntologyRestoreNode)
