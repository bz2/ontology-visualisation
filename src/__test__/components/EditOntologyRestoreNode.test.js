import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EditOntologyRestoreNode from '../../components/EditOntologyRestoreNode'

const setup = ({
  type,
  optionNodes,
  optionEdges
}) => {
  const props = {
    type,
    operation: 'restore',
    setStoreState: jest.fn(),
    addNumber: jest.fn(),
    optionNodes,
    optionEdges
  }

  const component = shallow(<EditOntologyRestoreNode {...props} />)

  return {
    component,
    props
  }
}

describe('EditOntologyRestoreNode', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot when type node and no options', () => {
    const {
      component
    } = setup({
      type: 'node',
      optionNodes: [],
      optionEdges: [],
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when type edge and no options', () => {
    const {
      component
    } = setup({
      type: 'edge',
      optionNodes: [],
      optionEdges: [],
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when type node and options', () => {
    const {
      component
    } = setup({
      type: 'node',
      optionNodes: [{
        value: 'http://test.test/test',
        label: 'Test'
      }],
      optionEdges: [],
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
