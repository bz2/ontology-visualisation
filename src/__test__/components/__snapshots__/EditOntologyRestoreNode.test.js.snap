// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditOntologyRestoreNode should match snapshot when type edge and no options 1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  addToArray={[MockFunction]}
  addToObject={[MockFunction]}
  mapToProps={[Function]}
  opearation="restore"
  optionEdges={Array []}
  optionNodes={Array []}
  removeFromObject={[MockFunction]}
  setStoreState={[MockFunction]}
  type="edge"
>
  <Component />
</Connect>
`;

exports[`EditOntologyRestoreNode should match snapshot when type node and no options 1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  addToArray={[MockFunction]}
  addToObject={[MockFunction]}
  mapToProps={[Function]}
  opearation="restore"
  optionEdges={Array []}
  optionNodes={Array []}
  removeFromObject={[MockFunction]}
  setStoreState={[MockFunction]}
  type="node"
>
  <Component />
</Connect>
`;

exports[`EditOntologyRestoreNode should match snapshot when type node and options 1`] = `
<Connect
  actions={
    Object {
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  addToArray={[MockFunction]}
  addToObject={[MockFunction]}
  mapToProps={[Function]}
  opearation="restore"
  optionEdges={Array []}
  optionNodes={
    Array [
      Object {
        "label": "Test",
        "value": "http://test.test/test",
      },
    ]
  }
  removeFromObject={[MockFunction]}
  setStoreState={[MockFunction]}
  type="node"
>
  <Component />
</Connect>
`;
